services/app-api/src/postgres/contractAndRates/approveContract.ts

```diff
@@ -0,0 +1,67 @@
+import { findContractWithHistory } from './findContractWithHistory'
+import { NotFoundError } from '../postgresErrors'
+import type { ContractType } from '../../domain-models/contractAndRates'
+import type { PrismaTransactionType } from '../prismaTypes'
+
+async function approveContract(
+    tx: PrismaTransactionType,
```

Comment:
This is tricky. The way you are calling this from the resolver you are not passing a transaction directly into this function. If you consolidate the db calls into one, there's no need for a transaction so if you change this to `client: PrismaClient` that would work fine, we do that in other single-call db handlers. If you keep the two db calls that need to be in a transaction, I suggest you model it after a more complex handler, you can look at `submitRate.ts` for an example. The model I've used is to have a separate very small function that takes `client: PrismaClient` that then calls a `approveContractInsideTransaction` function that has a tx. You setup the transaction in the calling function to make sure we don't accidentally return an error inside the transaction and commit a transaction that actually has an error. 

LMK if you want to chat about this one I got bit by this bug and breaking it out this way is more defensive.